docker-compose.yml

```diff
@@ -330,6 +330,8 @@ services:
       ARROW_USE_LD_GOLD: "ON"
       BUILD_WARNING_LEVEL: "PRODUCTION"
       ARROW_CTEST_TIMEOUT: 500
+      # GH-39973: Do not use debug memory pool for valgrind
+      ARROW_DEBUG_MEMORY_POOL: "none"
```

Comment:
Could you keep this list in alphabetical order?

Reply:
Ditto.